roles/setup_dbt2/tasks/dbt2_update_pg_hba.yml

```diff
@@ -1,56 +1,55 @@
 ---
-- name: Gather DBT-2 client addresses
-  ansible.builtin.set_fact:
-    dbt2client_item: >-
-      {
-        'users': 'all',
-        'databases': 'all',
-        'contype': 'host',
-        'source': '{{ hostvars[item]['private_ip'] }}/32',
-        'method': 'trust'
-      }
-  with_items: "{{ groups['dbt2_client'] }}"
-  when: "'dbt2_client' in groups"
-  register: dbt2client_list
+# Cannot use manage_dbserver.manage_hba_conf because it uses
+# community.postgresql.postgresql_pg_hba, which does not let us create a host
+# and hostssl line for the same source because contype is not part of the
+# unique identifier that community.postgresql.postgresql_pg_hba uses.
 
-- name: Gather DBT-2 driver addresses
-  ansible.builtin.set_fact:
-    dbt2driver_item: >-
-      {
-        'users': 'all',
-        'databases': 'all',
-        'contype': 'host',
-        'source': '{{ hostvars[item]['private_ip'] }}/32',
-        'method': 'trust'
-      }
-  with_items: "{{ groups['dbt2_driver'] }}"
-  register: dbt2driver_list
+- name: Find the hba file location
+  community.postgresql.postgresql_query:
+    login_user: "{{ pg_owner }}"
+    port: "{{ pg_port }}"
+    login_unix_socket: "{{ pg_unix_socket_directories[0] }}"
+    query: "show hba_file;"
+    db: "{{ pg_database }}"
+  become: true
+  become_user: "{{ pg_owner }}"
+  no_log: "{{ disable_logging }}"
+  register: hba_info
 
-- name: Make a list of DBT-2 client addresses
-  ansible.builtin.set_fact:
-    hba_entries: "{{ dbt2client_list.results | map(attribute='ansible_facts.dbt2client_item') | list }}"
+- name: Update pg_hba.conf with DBT-2 client host connections
+  ansible.builtin.lineinfile:
+    path: "{{ hba_info.query_result[0].hba_file }}"
+    line: "host all all {{ hostvars[item]['private_ip'] }}/32 trust"
+    create: true
+  with_items: "{{ groups['dbt2_client'] }}"
+  become: true
+  become_user: "{{ pg_owner }}"
```

Comment:
for benchmarking, does the use of `no_log: "{{ disable_logging }}"` matter at all? If so, I would recommend adding here, otherwise I'm okay with not having it

Reply:
for benchmarking, no.  for consistency with how we do things in the other roles, i will add momentarily.

Reply:
forced pushed an update